packages/astro/test/units/env/env-validators.test.js

```diff
@@ -0,0 +1,228 @@
+		});
+
+		it('Should not take a default value is the variable is not missing', () => {
+			const input = fixture.createInput('abc', {
+				type: 'string',
+				default: 'def',
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeValid('abc');
+		});
+	});
+
+	describe('number field', () => {
+		it('Should fail if the variable is missing', () => {
+			const input = fixture.createInput(undefined, {
+				type: 'number',
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeInvalid();
+		});
+
+		it('Should fail is the variable type is incorrect', () => {
+			const input = fixture.createInput('abc', {
+				type: 'number',
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeInvalid();
+		});
+
+		it('Should accept integers', () => {
+			const input = fixture.createInput('12345', {
+				type: 'number',
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeValid(12345);
+		});
+
+		it('Should accept floats', () => {
+			const input = fixture.createInput('12.34', {
+				type: 'number',
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeValid(12.34);
+		});
+
+		it('Should not fail if the optional variable is missing', () => {
+			const input = fixture.createInput(undefined, {
+				type: 'number',
+				optional: true,
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeValid(undefined);
+		});
+
+		it('Should not fail if the variable is missing and a default value is provided', () => {
+			const input = fixture.createInput(undefined, {
+				type: 'number',
+				default: 123,
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeValid(123);
+		});
+
+		it('Should not take a default value is the variable is not missing', () => {
+			const input = fixture.createInput('123', {
+				type: 'number',
+				default: 456,
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeValid(123);
+		});
+	});
+
+	describe('boolean field', () => {
+		it('Should fail if the variable is missing', () => {
+			const input = fixture.createInput(undefined, {
+				type: 'boolean',
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeInvalid();
+		});
+
+		it('Should fail is the variable type is incorrect', () => {
+			const input = fixture.createInput('abc', {
+				type: 'boolean',
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeInvalid();
+		});
+
+		it('Should not fail if the optional variable is missing', () => {
+			const input = fixture.createInput(undefined, {
+				type: 'boolean',
+				optional: true,
+			});
+			fixture.givenInput(input);
+			fixture.whenValidating();
+			fixture.thenResultShouldBeValid(undefined);
```

Comment:
I would change the fixture, considering how you're using them:
```js
      fixture.createInput(undefined, {
				type: 'boolean',
				optional: true,
			});
		  fixture.thenResultShouldBeValid(undefined);
```

The input you create is always the input that you pass in `.giveInput`. 

The `whenValidating()` can be part of `thenResultShouldBeValid` and `thenResultShouldBeInvalid`

Reply:
I was trying to follow arrange/act/assert but that makes sense, I'll update it